### PR TITLE
zkCheckout docs

### PR DESCRIPTION
**Pull request by StanislavBreadless**
There are 3 pages:

- `Getting started` (very similar to tutorials for other SDKs)
- `Reference`, slightly modified original docs
- `Appendix A`. the same purpose as the one for zksync.js

### CODE REVIEW

**Comment by StanislavBreadless** [comment]
The CI does not pass because of formatting errors in the files not related to this PR.

It's better to wait until the [PR with the new FAQ](https://github.com/matter-labs/zksync-docs/pull/45) is merged, before merging this one.

File: docs/sdk/checkout/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# zkSync checkout
+
+This section contains documentation for the zkCheckout SDK, the tool that helps conveniently create payments with
```
Comment (barakshani) [inline docs/sdk/checkout/README.md]: I think is should be "helps creating" or "helps to create", so either:
- helps conveniently creating
- helps to conveniently create (sounds better to me)

File: docs/sdk/checkout/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+zkSync.
+
+## Examples
+
+Some of the great examples of using the library:
```
Comment (barakshani) [inline docs/sdk/checkout/README.md]: I'd say "Great examples of the library uses" (without "some of")

File: docs/sdk/checkout/reference.md [hunk 3]
```diff
@@ -0,0 +1,354 @@
+# Reference
+
+## Interface overview
+
+The library provides a convenient way to integrate zkSync checkout flow into any website.
+
+The main export of the library is `CheckoutManager` class, which must be instantiated separately for each checkout.
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: is _the_ `CheckoutManager` class

File: docs/sdk/checkout/reference.md [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+
+The library also exports `Constants`, `Types`, `Utils`. These export constants, type definitions and type guards,
+collection of the utilities.
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: I don't understand the meaning of the second sentence.
Looks like we want to say "there _are_ export.... type guards _and_ collection of.."

File: docs/sdk/checkout/reference.md [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+
+It also exports functions that might be commonly used:
+
+- `estimateBatchFee` is used to estimate the fee for a batch of transactions.
+- `wait` is used to wait until a batch of transactions is either committed or verified.
+- `getState` is used to get the state of a zkSync account.
+- `checkEnoughBalance` is used to check if the user has enough balance to proceed to the checkout.
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: this is fine, but I'd rather say _sufficient_ than _enough_ in this context

File: docs/sdk/checkout/reference.md [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+  description?: string;
+}
+```
+
+- `from` field contains the Ethereum address of the transaction sender. The field can be omitted since the user has to
+  log in to zkSync before continuing the checkout process. But if this field is specified and the account from which the
+  user logs in to zkSync differs from any of the `from` fields of the transactions, the zkSync checkout page will notify
+  the user about that and will ask her to log in to the appropriate account.
+- `to` field contains the Ethereum address of the recipient of the transaction.
+- `token` — a the symbol of the token in which the transaction will be done.
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: "a the" --> "the"
We can say "The token symbol in which..."

File: docs/sdk/checkout/reference.md [hunk 7]
```diff
@@ -0,0 +1,354 @@
...
+  from?: string;
+  to: string;
+  token: TokenLike;
+  amount: string;
+  semanticType?: 'Transaction' | 'FeeOrCommission';
+  description?: string;
+}
+```
+
+- `from` field contains the Ethereum address of the transaction sender. The field can be omitted since the user has to
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: _The_ 'from' field contains.

If you prefer that 'from' is the first word, and for consistency with the items below:
'from' — this field contains / a field that contains

File: docs/sdk/checkout/reference.md [hunk 8]
```diff
@@ -0,0 +1,354 @@
...
+  log in to zkSync before continuing the checkout process. But if this field is specified and the account from which the
+  user logs in to zkSync differs from any of the `from` fields of the transactions, the zkSync checkout page will notify
+  the user about that and will ask her to log in to the appropriate account.
+- `to` field contains the Ethereum address of the recipient of the transaction.
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: see above

File: docs/sdk/checkout/reference.md [hunk 9]
```diff
@@ -0,0 +1,354 @@
...
+- `token` — a the symbol of the token in which the transaction will be done.
+- `amount` — amount of tokens in wei to be transferred.
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: amount of tokens, in wei, to be transferred

File: docs/sdk/checkout/reference.md [hunk 10]
```diff
@@ -0,0 +1,354 @@
...
+- `semanticType` — semantic type of the transaction, it is used to help the zkSync checkout popup provide better UX. For
+  instance, if your website collects fees in a certain transaction, you should set `semanticType` to
+  `'FeeOrCommission'`. The default value is `'Transaction'`.
+
+**Note** that if you set `semanticType` to `'FeeOrCommission'` then you also MUST provide a non-empty `description`. If
+you don't, the checkout will fail with `RECEIVED_INVALID_DATA_ERROR`.
+
+- `description` — the description of the transaction. For now, it is only used when `semanticType` is set to
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: "the description of the transaction" --> "transaction description"

File: docs/sdk/checkout/reference.md [hunk 11]
```diff
@@ -0,0 +1,354 @@
...
+  `'FeeOrCommission'`, and is ignored otherwise.
+
+### TokenLike
+
+Just a string. Has a special type name to clarify that the string should be equal to some token symbol.
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: I don't understand what this is for. Consider giving an explanation.

File: docs/sdk/checkout/reference.md [hunk 12]
```diff
@@ -0,0 +1,354 @@
...
+  feeToken: TokenLike,
+  userAddress?: string
+): Promise<string[]>
+```
+
+Gets an array of the zkSync transactions for the flow and token in which the zkSync fee will be paid. Also,
+`userAddress` is the optional parameter, which, if supplied, will be used to check that the user logs in with the
+correct address.
+
+Returns a Promise that resolves with the array of the hashes of the transactions in the same order as the transactions
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: promise? why capital p?

File: docs/sdk/checkout/reference.md [hunk 13]
```diff
@@ -0,0 +1,354 @@
...
+were supplied.
+
+**Note, that you should NOT add the fee transaction on yourselves.** Just specify the `feeToken` and everything else
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: "on yourselves" --> "by yourselves".
I'd use _yourself_ instead.

File: docs/sdk/checkout/reference.md (hunk 12 above)
Comment (barakshani) [reply docs/sdk/checkout/reference.md]: few lines below it says "promise"

File: docs/sdk/checkout/reference.md [hunk 14]
```diff
@@ -0,0 +1,354 @@
...
+  network: EthereumNetwork
+): Promise<string>;
+```
+
+Gets an array of the zkSync transactions for the flow and the token in which the zkSync fee will be paid.
+
+Returns the promise that resolves with the number of `feeToken` (in wei) that will be paid by the user.
+
+Note, that the value that the method returns does not take into account the amount of fee that will be paid by the user
+in process of depositing the tokens she lacks. But it DOES take into account the fact that one transaction will be added
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: "in _the_ process of depositing..."

Comment (barakshani) [reply docs/sdk/checkout/reference.md]: what do we mean by "tokens she lacks"? some context seems to be missing

Comment (barakshani) [reply docs/sdk/checkout/reference.md]: Generally, the first sentence is quite long to follow, and the second one is not so clear.

File: docs/sdk/checkout/reference.md [hunk 15]
```diff
@@ -0,0 +1,354 @@
...
+async function checkEnoughBalance(
+  transactions: ZkSyncTransaction[],
+  feeToken: string,
+  senderAddress: string,
+  ethProvider: ethers.providers.BaseProvider,
+  network: EthereumNetwork
+): Promise<boolean>;
+```
+
+Gets the list of transactions, token to pay the fee with, and the address of the user as parameters. Also takes an
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: "Gets the list" --> "Gets a list"

File: docs/sdk/checkout/reference.md [hunk 16]
```diff
@@ -0,0 +1,354 @@
...
+ethers provider as a parameter. It is needed to check L1 balance of the user.
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: "It is needed to check L1 balance of the user"
What is needed, the ethers provider? if so, I'd say
"It is needed for checking the L1 balance of the user."

File: docs/sdk/checkout/reference.md [hunk 17]
```diff
@@ -0,0 +1,354 @@
...
+
+Returns a Promise that resolves with `true` if the user has enough combined balance on L1 + L2 to do all the
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: Promise or promise?

File: docs/sdk/checkout/reference.md [hunk 18]
```diff
@@ -0,0 +1,354 @@
...
+transactions and pay the fee, and `false` otherwise.
+
+The function can throw `FAILED_TO_GET_ONCHAIN_BALANCE` if it fails to get the on-chain balance of the user.
+
+## Constants
+
+A collection of all the important constants.
+
+The majority of the constants are for the internal use but there are some of them that might be useful.
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: "are for the internal use" --> "are for internal use,"

"but there are some of them that" --> ", but some of them might"

**Review by barakshani — COMMENTED** [review]
There are a few typos, the rest are suggestion for your consideration.

(I will continue the review later)

File: docs/sdk/checkout/reference.md [hunk 19]
```diff
@@ -0,0 +1,354 @@
...
+} as const;
+```
+
+### Client-side errors vs zkSync-side errors
+
+The `CheckoutErrors` contains the errors of which might happen both on the zkSync side and on the client's side.
+
+#### Common errors
+
+- `NOT_IMPLEMENTED_ERROR`. When the library is complete, there should be no places, where such an error could be thrown,
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: "there should be no places where such an error could be thrown" (without the comma)

File: docs/sdk/checkout/reference.md [hunk 20]
```diff
@@ -0,0 +1,354 @@
...
+  but for the time of development, this error is thrown in the places which have not been properly implemented yet.
+- `LOGICAL_ERROR`. This error is thrown if something that should never happen happens. That means that there is a bug in
+  the library's code.
+- `INVALID_SENDER_ERROR`. This error is thrown if the `userAddress` argument (if supplied) of the `zkSyncBatchCheckout`
+  function is different from any of the supplied `from` fields of the transactions, or if any supplied `from` field of
+  the transactions is different from any other. In other words, that error is thrown if the consistency of the sender
+  address is broken.
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: I would reorder the sentences:
"This error is thrown if the consistency of the sender address is broken. That is, if either ......, or if any supplied..."

File: docs/sdk/checkout/reference.md [hunk 21]
```diff
@@ -0,0 +1,354 @@
...
+- `FAILED_TO_GET_ONCHAIN_BALANCE`. This error is thrown if the library fails to retrieve the user's Ethereum balance.
+
+#### Client-side errors
+
+- `NOT_IN_BROWSER_ERROR`. This error is thrown because the library is intended to be used only in the browser
+  environment. Particularly, the error is thrown if the `window` global object is not defined.
+- `OPENING_TIMEOUT_ERROR`. This error is thrown if the zkSync page takes too long to load. Used to tackle the case when
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: "Used to tackle" --> "It is used to tackle"

File: docs/sdk/checkout/reference.md [hunk 22]
```diff
@@ -0,0 +1,354 @@
...
+  the user opens the checkout page, but then closes it quickly.
+- `FAILED_TO_OPEN_ZKSYNC`. This error is thrown if the client's page fails to open the zkSync page. (Technically
+  speaking, when the `window.open` returns null)
+- `ZKSYNC_CLOSED_ERROR`. This error is thrown when the user closes the zkSync checkout page without initiating any
+  transactions.
+
+#### zkSync-side errors
+
+- `NO_OPENER_ERROR`. This error is thrown if the checkout page is opened without an opener (basically that means, that
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: "basically that means, that the page was opened directly" --> "basically this means that..."

File: docs/sdk/checkout/reference.md [hunk 23]
```diff
@@ -0,0 +1,354 @@
...
+  the page was opened directly, but not as a popup). That's why you may see the gray screen if you open the checkout
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: "the gray screen" --> "a gray screen" ?

File: docs/sdk/checkout/reference.md [hunk 24]
```diff
@@ -0,0 +1,354 @@
...
+  page directly in the browser.
+- `CHECKOUT_NOT_STARTED_ERROR`. This error is thrown if the checkout page tries to retrieve the state (e,g, list of
+  transactions, fee token, etc) of the checkout before the data has been received.
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: e,g, --> e.g.
etc --> etc.

File: docs/sdk/checkout/reference.md [hunk 25]
```diff
@@ -0,0 +1,354 @@
...
+## Protocol
+
+Due to modern web-browsers' security measures, the opening of the tab with the checkout flow internally might look like
+a connection establishment via an exchange of `postMessage`-s.
+
+The protocol follows
+[web security best practices](https://developer.mozilla.org/en-US/docs/Web/library/Window/postMessage#Security_concerns),
+associated with cross-origin communication.
+
+The internal part of the checkout will consist of the following stages:
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: "will consist" --> "consists"?

File: docs/sdk/checkout/reference.md [hunk 26]
```diff
@@ -0,0 +1,354 @@
...
+
+1. The client calls `zkSyncBatchCheckout`.
+2. When the tab is open, the zkSync page will send the `ZKSYNC_OPENED` message. If it fails to do within
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: " If it fails to do within" --> " If it fails to do _so_ within"

File: docs/sdk/checkout/reference.md [hunk 27]
```diff
@@ -0,0 +1,354 @@
...
+   `OPENING_TIMEOUT` ms, the checkout Promise resolves with `OPENING_TIMEOUT_ERROR`. After receiving the `ZKSYNC_OPEN`
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: again "Promise" or "promise"? This page uses both

File: docs/sdk/checkout/reference.md [hunk 28]
```diff
@@ -0,0 +1,354 @@
...
+   message, the client's page will send a `START_SESSION` message to the zkSync page. The message contains:
+   - Array of the transactions, specified by the user.
+   - Token to pay the fee with.
+   - The origin of the opener. Must be equal to the origin of the opener (zkSync checks it on its side).
+   - The checkout id — a feature that enables multiple checkouts at the same time.
+3. If at least in one transaction the `from` field differs from the user's one, the zkSync page will tell the user about
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: differs from the user's one --> from the user's address ?

File: docs/sdk/checkout/reference.md [hunk 29]
```diff
@@ -0,0 +1,354 @@
...
+   it and will offer him to log in with the other account.
+
+The zkSync page will remember the client's origin. Now, all the `postMessage`-s sent from the zkSync page will only go
+to the remembered origin. 4. Every `CLOSE_POLL_INTERVAL` ms the client's page will check if the zkSync page is closed.
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: new line for 4.

File: docs/sdk/checkout/reference.md (hunk 4 above)
Comment (StanislavBreadless) [reply docs/sdk/checkout/reference.md]: `export` is a verb here, no need for `are`.

I propose:

`type guards and collection of` => `as well as some utility functions`

File: docs/sdk/checkout/reference.md (hunk 6 above)
Comment (StanislavBreadless) [reply docs/sdk/checkout/reference.md]: 1. :+1: 
2. Symbol is a property of a token, so I think it's fine.

File: docs/sdk/checkout/reference.md (hunk 9 above)
Comment (StanislavBreadless) [reply docs/sdk/checkout/reference.md]: Not sure why

File: docs/sdk/checkout/reference.md (hunk 10 above)
Comment (StanislavBreadless) [reply docs/sdk/checkout/reference.md]: Thanks! I guess "the transaction description" is better

File: docs/sdk/checkout/reference.md (hunk 11 above)
Comment (StanislavBreadless) [reply docs/sdk/checkout/reference.md]: What exactly do you find hard to understand?

File: docs/sdk/checkout/reference.md (hunk 17 above)
Comment (StanislavBreadless) [reply docs/sdk/checkout/reference.md]: Grammarly recommends me to write `Promise`, since `Promise` is actually a JS built-in class for async functions.

However, [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises) I can see that unless explicitly talking about the class itself, it is fine to use lowercase.

Comment (StanislavBreadless) [reply docs/sdk/checkout/reference.md]: I'll go with lowercase

File: docs/sdk/checkout/reference.md [hunk 30]
```diff
@@ -0,0 +1,354 @@
...
+
+If the zkSync page is closed without starting the transactions, the client waits for `FOLLOWUP_HASHES_TIMEOUT` ms in
+case the hashes were sent to the client, but the browser was too slow. If the hashes don't arrive within this time, the
+checkout promise will be rejected with `ZKSYNC_CLOSED_ERROR`.
+
+On the other hand, every `OPENER_CLOSED_POLL_INTERVAL` ms the zkSync page checks if the client's page is closed. If it
+is closed before the transactions' hashes were sent, there is a danger that the hashes that the zkSync page will send
+won't be processed by the client. To prevent this from happening the zkSync page will warn the user about that and
+advice her not to continue the checkout. 5. When the user clicks on transfer, the zkSync page sends the opener the array
+of hashes of the transactions. 6. (Optional for the client) After the client receives the transactions, he can either
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: new lines for 5 & 6

File: docs/sdk/checkout/reference.md [hunk 31]
```diff
@@ -0,0 +1,354 @@
...
+to the remembered origin. 4. Every `CLOSE_POLL_INTERVAL` ms the client's page will check if the zkSync page is closed.
+
+If the zkSync page is closed without starting the transactions, the client waits for `FOLLOWUP_HASHES_TIMEOUT` ms in
+case the hashes were sent to the client, but the browser was too slow. If the hashes don't arrive within this time, the
+checkout promise will be rejected with `ZKSYNC_CLOSED_ERROR`.
+
+On the other hand, every `OPENER_CLOSED_POLL_INTERVAL` ms the zkSync page checks if the client's page is closed. If it
+is closed before the transactions' hashes were sent, there is a danger that the hashes that the zkSync page will send
+won't be processed by the client. To prevent this from happening the zkSync page will warn the user about that and
+advice her not to continue the checkout. 5. When the user clicks on transfer, the zkSync page sends the opener the array
```
Comment (barakshani) [inline docs/sdk/checkout/reference.md]: "sends the opener the array" 
A bit confusing, so maybe "sends to the opener the array"

File: docs/sdk/checkout/reference.md (hunk 4 above)
Comment (barakshani) [reply docs/sdk/checkout/reference.md]: oh, I see : ) so: "This exports..." I believe, but I would say "It" instead.

File: docs/sdk/checkout/reference.md (hunk 9 above)
Comment (barakshani) [reply docs/sdk/checkout/reference.md]: mmm....... it feels more natural to me.
Basically we want to say "amount of tokens to be transferred", so the "in wei" is a short clause that provides more information but not really necessary.
BTW, below we have "(in wei)", but to be fair it is after the "in wei" is stated here already, so it makes sense to put it in parentheses.

Anyway, whatever you like. Your call.

File: docs/sdk/checkout/reference.md (hunk 11 above)
Comment (barakshani) [reply docs/sdk/checkout/reference.md]: If I understand correctly here we explain what this field "TokenLike" is.
First thing is that it's not clear to me why we even have this field (please keep in mind that I don't really have context -- people who will read it, will most likely have context).
Secondly, what do we mean by "some token field"? Shouldn't we clarify if it's a specific token symbol?

File: docs/sdk/checkout/reference.md (hunk 4 above)
Comment (StanislavBreadless) [reply docs/sdk/checkout/reference.md]: I guess `They` is better here since the list is about what do objects `Constants`, `Types` and `Utils` export.

File: docs/sdk/checkout/reference.md (hunk 9 above)
Comment (StanislavBreadless) [reply docs/sdk/checkout/reference.md]: Grammarly says it is fine, I'll go without commas

File: docs/sdk/checkout/reference.md (hunk 11 above)
Comment (StanislavBreadless) [reply docs/sdk/checkout/reference.md]: Every token has a symbol. Like 'USDC' for instance. I can say `... should be equal to a token's symbol.`

File: docs/sdk/checkout/reference.md (hunk 24 above)
Comment (StanislavBreadless) [reply docs/sdk/checkout/reference.md]: 1. :+1: 
2. Not so sure, since there is a bracket after that

File: docs/sdk/checkout/reference.md (hunk 27 above)
Comment (StanislavBreadless) [reply docs/sdk/checkout/reference.md]: Locally I don't have that :eyes:

File: docs/sdk/checkout/reference.md (hunk 30 above)
Comment (StanislavBreadless) [reply docs/sdk/checkout/reference.md]: Big thanks

**Comment by StanislavBreadless** [comment]
@barakshani The reference was has been split into different pages, so it is consistent with the docs of the JS SDK

File: docs/sdk/checkout/reference.md (hunk 11 above)
Comment (barakshani) [reply docs/sdk/checkout/reference.md]: "a token's symbol" means that _any_ symbol can be given here - so theoretically a user can use 'USD coin', but then specify the symbol 'USDT' (instead of 'USDC'). It should be more correct to say "the token's symbol".

I would rephrase this sentence to something like "This is a string that should equal the token's symbol".

File: docs/sdk/checkout/reference.md (hunk 24 above)
Comment (barakshani) [reply docs/sdk/checkout/reference.md]: I thought that " etc. " is atomic, because it is an abbreviation.
Turns out " etc " is also used by itself, e.g. https://en.wikipedia.org/wiki/Et_cetera

File: docs/sdk/checkout/reference.md (hunk 27 above)
Comment (barakshani) [reply docs/sdk/checkout/reference.md]: I meant the entire page (reference.md ) not this specific section -- so referring to previous places I mentioned that : /

File: docs/sdk/checkout/tutorial.md [hunk 32]
```diff
@@ -0,0 +1,194 @@
...
+
+**Please note, that the library works only in browser environment, since the user is guided through zkCheckout via a
+popup**
+
+## Structure of the tutorial
+
+This tutorial does not cover back-end or front-end code that is needed for a website but focuses solely on how to use
+the `zksync-checkout` library.
+
+This tutorial will show the basic flow of the checkout by putting all the action inside the single function:
```
Comment (barakshani) [inline docs/sdk/checkout/tutorial.md]: 1. action --> actions

2. the basic flow of the checkout --> "the basic checkout flow" sounds better (to me)

3. Maybe I'm wrong here, but "the single function" doesn't sound right. I'd say either:
- the single function `checkoutUser':
- a single function:

File: docs/sdk/checkout/tutorial.md [hunk 33]
```diff
@@ -0,0 +1,194 @@
...
+  feeToken: Types.TokenLike,
+  // The app can accept payments whether you know the address of the user or you don't
+  address?: string
+) {
+  // Here will go the code
+}
+```
+
+Let's say that the user has to make two payments: one for some apples worth 23 DAI and one for the bananas worth 55.5
+DAI, the user should pay the zkSync fee also in DAI. Then you could call the function the following way:
```
Comment (barakshani) [inline docs/sdk/checkout/tutorial.md]: replace comma with period:
DAI. The user should pay...

File: docs/sdk/checkout/tutorial.md [hunk 34]
```diff
@@ -0,0 +1,194 @@
...
+
+// ..
+
+// 'ropsten' and 'rinkeby' are also supported
+const checkoutManager = new CheckoutManager('mainnet');
+```
+
+## Checking user's balance
+
+If the address of the user is known, we should check if the user has sufficient funds to proceed with the payment. This
```
Comment (barakshani) [inline docs/sdk/checkout/tutorial.md]: If the address of the user --> If the user's address ?

**Review by barakshani — APPROVED** [review]

File: docs/sdk/checkout/reference.md (hunk 11 above)
Comment (StanislavBreadless) [reply docs/sdk/checkout/reference.md]: This is a description of a type, so I guess "a" is totally fine here. We do not talk about any specific token.

File: docs/sdk/checkout/reference.md (hunk 27 above)
Comment (StanislavBreadless) [reply docs/sdk/checkout/reference.md]: There should be "promise", but by the time your comment was written, the "Promise" had been removed from the `referene.md`. That's why I was wondering how did you find the "Promise" with capital P :slightly_smiling_face:

File: docs/sdk/checkout/tutorial.md (hunk 32 above)
Comment (StanislavBreadless) [reply docs/sdk/checkout/tutorial.md]: 1. "all the action" is a stale phrase, so it is fine. You can find more examples by googling "put all the action into"
2. :+1: 
3. :+1:

File: docs/sdk/checkout/tutorial.md (hunk 33 above)
Comment (StanislavBreadless) [reply docs/sdk/checkout/tutorial.md]: I will put "also" right after the comma, so now it looks like this:

Let's say that the user has to make two payments: one for some apples worth 23 DAI and one for the bananas worth 55.5
DAI, also the user should pay the zkSync fee in DAI.

So that it is seen that these 3 statements have nothing to do with each other (the user could pay the fee in another token if the owner of the website wanted to)
